apps/bare-expo/ios/Podfile

```diff
@@ -6,9 +6,9 @@ platform :ios, '12.0'
 inhibit_all_warnings!
 
 # Import the auto-linking packages
-require File.join(`node --print "require.resolve('react-native/package.json')"`, "../scripts/react_native_pods")
-require File.join(`node --print "require.resolve('@react-native-community/cli-platform-ios/package.json')"`, "../native_modules")
-require File.join(`node --print "require.resolve('expo-modules-core/package.json')"`, "../scripts/autolinking")
+require File.join(File.dirname(`node --print "require.resolve('react-native/package.json')"`), "scripts/react_native_pods")
```

Comment:
Might be nice to have a common method that we import so this logic is hidden away from user-space and we could make it handle errors a bit better if needed.

```
requireFromNodeModule("react-native", "./scripts/react_native_pods")
```

Reply:
the question is where to put the `requireFromNodeModule`. if we deploy this in npm and node_modules, this is yet another workspace support issue. if we define `requireFromNodeModule` in Podfile, that requires many patches in Podfile for migration or be risky to update by config plugin.

so far my thought is about to add a ruby file in the same directory with Podfile and put all the logic there.
we should reduce the chances to update the ruby file as possible b/c it's hard to make sure user's ruby file be updated.
also had some experimented idea to monkey patch `require_relative` and showing here:

```diff
--- a/apps/bare-expo/ios/Podfile
+++ b/apps/bare-expo/ios/Podfile
 platform :ios, '12.0'
 inhibit_all_warnings!

+require_relative 'node_modules_require_helper'
+
 # Import the auto-linking packages
require_relative '../node_modules/react-native/scripts/react_native_pods'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'
require_relative '../node_modules/expo-modules-core/scripts/autolinking'
+# There two styles are also supported.
+# require_relative_node_modules '../node_modules/expo-modules-core/scripts/autolinking'
+# require_from_node_modules 'expo-modules-core', 'scripts/autolinking'
```

```ruby
# node_modules_require_helper.rb
require 'open3'

def require_from_node_module(module_name, script)
  require File.join(File.dirname(`node --print "require.resolve('#{module_name}/package.json')"`), script)
end

def find_absolute_node_module_dir(module_name)
  stdout, _, _ = Open3.capture3("node --print \"require.resolve('#{module_name}/package.json')\"")
  unless stdout.strip.empty?
    return File.dirname(stdout)
  end
  return nil
end

def find_absolute_node_module_path(relative_node_module_path)
  relative_node_module_path.match(/\/node_modules\/(.+)$/) do |matched|
    chunks = matched[1].split('/')
    for index in 1..chunks.length-1
      module_name = chunks[0..index-1].join('/')
      file_path = chunks[index..-1].join('/')
      absolute_dir = find_absolute_node_module_dir(module_name)
      return File.expand_path(file_path, absolute_dir) if absolute_dir
    end
  end
  return nil
end

# official method
# usage: `require_from_node_modules('expo_modules_core', 'scripts/autolinking')`
def require_from_node_modules(module_name, file_path)
  absolute_dir = find_absolute_node_module_dir(module_name)
  if absolute_dir
    require File.expand_path(file_path, absolute_dir)
  end
end

# official method 2: pretty much like `require_relative` but not monkey patching
# usage: `require_relative_node_modules '../node_modules/expo-modules-core/autolinking'`
def require_relative_node_modules(name)
  absolute_node_module_path = find_absolute_node_module_path(name)
  if absolute_node_module_path
    require absolute_node_module_path
  end
end

# monkey patch require_relative to support node_modules
module Kernel
  define_method(:require_relative) do |name|
    absolute_node_module_path = find_absolute_node_module_path(name)
    if absolute_node_module_path
      require absolute_node_module_path
    else
      # after monkey patching `require_relative`, we cannot use original `require_relative`.
      # because relative path will reset as related to `node_modules_require_helper.rb`
      caller_dir = File.dirname(caller.first.split(/:\d/, 2).first)
      require File.expand_path(name, caller_dir)
    end
  end
end
```

Reply:
I think there is no better solution for that problem. Maybe this `require.resolve` thing looks weird, but since it's one-liner it's also pretty easy to understand what it does. Any other solution increases the technical debt.

Reply:
> I think there is no better solution for that problem. Maybe this `require.resolve` thing looks weird, but since it's one-liner it's also pretty easy to understand what it does. Any other solution increases the technical debt.

agree with that there are no better solutions so far. i am going to merge this pr. if there are any new idea, let's open new pr.